projects/js-packages/charts/src/charts/segmented-bar/segmented-bar.tsx

```diff
@@ -0,0 +1,314 @@
+import { useParentSize } from '@visx/responsive';
+import { useTooltip, useTooltipInPortal } from '@visx/tooltip';
+import clsx from 'clsx';
+import { useMemo, forwardRef, useCallback, useContext } from 'react';
+import {
+	GlobalChartsProvider,
+	GlobalChartsContext,
+	useGlobalChartsTheme,
+	useChartId,
+} from '../../providers';
+import styles from './segmented-bar.module.scss';
+import type { SegmentedBarProps, SegmentedBarSegment, SegmentedBarResponsiveConfig } from './types';
+import type { MouseEvent, FC } from 'react';
+
+const DEFAULT_WIDTH = 300;
+const DEFAULT_HEIGHT = 8;
+const DEFAULT_GAP = 0;
+const DEFAULT_BORDER_RADIUS = 4;
+const DEFAULT_LABEL_FORMATTER = ( value: number ) => value.toString();
+
+/**
+ * Normalizes input values to SegmentedBarSegment array.
+ * @param values      - Input values array (numbers or segment objects).
+ * @param colors      - Optional custom colors array.
+ * @param themeColors - Theme colors to use as fallback.
+ * @return Normalized array of SegmentedBarSegment objects.
+ */
+const normalizeSegments = (
+	values: number[] | SegmentedBarSegment[],
+	colors: string[] | undefined,
+	themeColors: string[]
+): SegmentedBarSegment[] => {
+	return ( values || [] ).map( ( value, index ) => {
+		const segment: SegmentedBarSegment = typeof value === 'number' ? { value } : { ...value };
+
+		// Apply color priority: segment.color > colors prop > theme colors
+		if ( ! segment.color ) {
+			segment.color = colors?.[ index ] || themeColors[ index % themeColors.length ] || '#000000';
+		}
+
+		return segment;
+	} );
+};
+
+const SegmentedBarComponent = forwardRef< HTMLDivElement, SegmentedBarProps >(
+	(
+		{
+			values,
+			mode = 'proportional',
+			colors,
+			marker,
+			showLabels = true,
+			width = DEFAULT_WIDTH,
+			height = DEFAULT_HEIGHT,
+			gap = DEFAULT_GAP,
+			borderRadius = DEFAULT_BORDER_RADIUS,
+			className,
+			chartId: providedChartId,
+			labelFormatter = DEFAULT_LABEL_FORMATTER,
+			withTooltips = false,
+		},
+		ref
+	) => {
+		const theme = useGlobalChartsTheme();
+		const generatedChartId = useChartId();
+		const chartId = providedChartId || generatedChartId;
+
+		const themeColors = useMemo( () => {
+			return theme?.colors || [ '#000000' ];
+		}, [ theme ] );
+
+		// Normalize values to segments
+		const segments = useMemo( () => {
+			return normalizeSegments( values, colors, themeColors );
+		}, [ values, colors, themeColors ] );
```

Comment:
For consistency with other charts this should use `getElementStyles` from the context:
```suggestion
		const { getElementStyles, theme } = useGlobalChartsContext();
		const generatedChartId = useChartId();
		const chartId = providedChartId || generatedChartId;

		// Normalize values to segments
		const segments = useMemo( () => {
			return values.map( ( value, index ) => {
				const segment: SegmentedBarSegment = typeof value === 'number' ? { value } : { ...value };

				segment.color = getElementStyles( {
					index,
					overrideColor: value.color || colors?.[ index ],
				} ).color;

				return segment;
			} );
		}, [ values, getElementStyles, colors ] );
```